frontend/src/pages/projects/screens/spawner/imageSelector/ImageVersionSelector.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { FormGroup, Select, SelectOption, Text, Title } from '@patternfly/react-core';
+import { ImageVersionSelectDataType, ImageVersionSelectOptionObjectType } from '../types';
+import {
+  checkTagBuildValid,
+  compareImageVersionOptionOrder,
+  getAvailableVersionsForImageStream,
+  getImageVersionDependencies,
+  getImageVersionSoftwareString,
+  isImageVersionSelectOptionObject,
+} from '../spawnerUtils';
+import ImageVersionTooltip from './ImageVersionTooltip';
+import { getDashboardMainContainer } from '../../../../../utilities/utils';
+
+type ImageVersionSelectorProps = {
+  data: ImageVersionSelectDataType;
+  selectedImageVersion?: ImageVersionSelectOptionObjectType;
+  onImageVersionSelect: (selection: ImageVersionSelectOptionObjectType) => void;
+};
+
+const ImageVersionSelector: React.FC<ImageVersionSelectorProps> = ({
+  selectedImageVersion,
+  onImageVersionSelect,
+  data,
+}) => {
+  const [versionSelectionOpen, setVersionSelectionOpen] = React.useState<boolean>(false);
+
+  const { imageStream, versionOptions, buildStatuses } = data;
+
+  if (!imageStream || getAvailableVersionsForImageStream(imageStream, buildStatuses).length <= 1) {
+    return null;
+  }
+
+  const options = [...versionOptions].sort(compareImageVersionOptionOrder).map((optionObject) => {
+    const imageVersion = optionObject.imageVersion;
+    // Cannot wrap the SelectOption with Tooltip because Select component requires SelectOption as the children
+    // Can only wrap the SelectOption children with Tooltip
+    // But in this way, you will only see the tooltip when you hover the option main text (excluding description), not the whole button
+    return (
+      <SelectOption
+        key={`${imageStream.metadata.name}-${imageVersion.name}`}
+        value={optionObject}
+        description={getImageVersionSoftwareString(imageVersion)}
+        isDisabled={!checkTagBuildValid(buildStatuses, imageStream, imageVersion)}
+      >
+        <ImageVersionTooltip dependencies={getImageVersionDependencies(imageVersion, false)}>
+          {optionObject.toString()}
```

Comment:
Uh... this is bizarre... almost never do you want to do `.toString()` as it often doesn't give you a nice printout. If this is truly an Object, do we not want to just use `JSON.stringify(obj, null, 2)` or something like that to space out the data into readable values? Or better yet, convert it to a readable list and not use JSON/Object outputs?

Reply:
This optionObject here already defined the `toString()` function when I created it, it's a SelectOption object, so we can call this function directly and get the result we want.

Reply:
Yup, failed to read the code properly... thanks for clarifying.